### PR TITLE
feat: #WB2-155, add Cookie Consent feature

### PR DESCRIPTION
**Pull request by david-cc**
# Description

Adding Cookie Consent through the Alert Component as a "Confirm box".

Also adding "usePreferences" hook to get and save user preferences easily.

## Which Package changed?

Please check the name of the package you changed

- [X] Components
- [ ] Core
- [ ] Icons
- [ ] Hooks

## Has the documentation changed?

- [ ] Storybook

## Type of change

Please check options that are relevant.

- [ ] Chore (PATCH)
- [ ] Doc (PATCH)
- [ ] Bug fix (PATCH)
- [X] New feature (MINOR)
- [ ] Breaking change (MAJOR)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

File: packages/react/src/core/usePreferences/usePreferences.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { odeServices } from "edifice-ts-client";
+
+export default function usePreferences(name: string): Function[] {
```
Comment (ccreusat) [inline packages/react/src/core/usePreferences/usePreferences.ts:3]: @david-cc t'es obligé de typer le retour du hook ? tu peux laisser ts gérer

File: packages/react/src/core/usePreferences/usePreferences.ts
Comment (ccreusat) [inline packages/react/src/core/usePreferences/usePreferences.ts:1]: 👍🏼

File: packages/react/src/core/OdeClientProvider/OdeClientProvider.tsx [hunk 2]
```diff
@@ -49,6 +53,9 @@ export interface ContextProps {
 export const OdeClientContext = createContext<ContextProps | null>(null!);
 
 export function OdeClientProvider({ children, params }: OdeClientProps) {
+  const [showCookiesConsent, setShowCookiesConsent] = useState<boolean>(false);
```
Comment (ccreusat) [inline packages/react/src/core/OdeClientProvider/OdeClientProvider.tsx:56]: @david-cc tu penses que t'as moyen d'extraire ça dans un hook `useCookie` quelque chose pour pas polluer `OdeClientProvider` ? Et appeler ici que ce qui est nécessaire?

File: packages/react/src/core/usePreferences/usePreferences.ts (hunk 1 above)
Comment (david-cc) [reply packages/react/src/core/usePreferences/usePreferences.ts:3]: ah ouai en effet j'avais un soucis de TS, mais c'était parce que je retournais un tableau au lieu d'un objet dans mon hook

File: packages/react/src/core/OdeClientProvider/OdeClientProvider.tsx (hunk 2 above)
Comment (david-cc) [reply packages/react/src/core/OdeClientProvider/OdeClientProvider.tsx:56]: yes en effet !

**Pull request merged**